### PR TITLE
NavigationViewItem input handling fixes

### PR DESCRIPTION
**Pull request by RBrid**
These changes address a series of bugs in the NavigationViewItem implementation:
#1 NavigationViewItem does not update its visual state when its IsSelected property changed.
-	In WinUI3, often when you tap a NavigationViewItem, it stays dark gray.
-	When you select a NavigationViewItem programmatically, it does not change its visual state.
Fix: Listening to IsSelected change notifications to refresh visual state.

#2 PonterMoved does not update NavigationViewItem visual state in WinUI3.
-	Move mouse cursor over NavigationViewItem A
-	Tap NavigationViewItem B
-	Move mouse cursor a bit --> NavigationViewItem A does not update its visual state.
Fix: Listening to PointerMoved to update visual state when pointer is not thought to be over.

#3 Mouse pointer is not captured in PointerPressed.
-	Mouse button down on NavigationViewItem.
-	Move mouse outside of NavigationViewItem.
-	Release mouse button --> NavigationViewItem does not update its NavigationViewItem visual state. It stays dark gray until you interact with it again.
Fix: Pointer is captured in PointerPressed and released in PointerReleased, or when pointer is lost.

#4 IsEnabled change does not update NavigationViewItem visual state.
Fix: Listening to NavigationViewItem.IsEnabledChanged to clear the state flags and refresh visual state.

#5 Beginning of OnApplyTemplate does not reset template parts, or clear event handlers.
Fix: All control parts and handler are cleared at beginning of OnApplyTemplate to avoid leftover incorrect fields.

#6 Pointer input handlers that reset internal flags are not invoked when args.Handled is true causing bugs.
Fix: Input handlers that reset flags are not skipped thanks to the handledEventsToo=True feature.

#7 Multi finger inputs are not handled properly. Removing second finger clears pressed state for instance.
Fix: Only one pointer Id is tracked for a given NavigationViewItem now. Any newcomer is ignored.

o	Also did some dead code cleanup, and code cleanup in general.
o	Did some ad-hoc testing, expanded controls test app and NavigationView test.

(These changes also address 26317023 once pushed into WinUI)

### CODE REVIEW

File: dev/NavigationView/NavigationViewItem.cpp [hunk 1]
```diff
@@ -27,15 +27,15 @@ static constexpr auto c_chevronHidden = L"ChevronHidden"sv;
 static constexpr auto c_chevronVisibleOpen = L"ChevronVisibleOpen"sv;
 static constexpr auto c_chevronVisibleClosed = L"ChevronVisibleClosed"sv;
 
-void NavigationViewItem::UpdateVisualStateNoTransition()
+NavigationViewItem::NavigationViewItem()
```
Comment (RBrid) [inline dev/NavigationView/NavigationViewItem.cpp:30]: Placed the constructor towards the top of the file.

File: dev/NavigationView/NavigationViewItem.h [hunk 2]
```diff
@@ -50,35 +49,37 @@ class NavigationViewItem :
...
+    bool IsRepeaterVisible() const;
     void PropagateDepthToChildren(int depth);
 
 private:
+    winrt::UIElement const GetPresenterOrItem() const;
+    NavigationViewItemPresenter* GetPresenter() const;
+
     void UpdateNavigationViewItemToolTip();
     void SuggestedToolTipChanged(winrt::IInspectable const& newContent);
-    void OnNavigationViewRepeaterPositionChanged() override;
```
Comment (RBrid) [inline dev/NavigationView/NavigationViewItem.h:76]: Renamed NavigationViewItemBasePosition for consistency with other names.

File: dev/TabView/TestUI/TabViewPage.xaml [hunk 3]
```diff
@@ -153,12 +153,14 @@
 
                         <StackPanel x:Name="FirstTabContent" AutomationProperties.Name="FirstTabContent">
                             <Button x:Name="FirstTabButton" AutomationProperties.Name="FirstTabButton" Margin="8" FontSize="20">Home Button</Button>
-                            <Button x:Name="TabViewSizingPageButton" AutomationProperties.Name="TabViewSizingPageButton"
-                                    Margin="8" Click="TabViewSizingPageButton_Click"
-                                    FontSize="20">TabView Sizing Page</Button>
-                            <Button x:Name="TabViewTabClosingBehaviorButton" AutomationProperties.Name="TabViewTabClosingBehaviorButton"
-                                    Margin="8" Click="TabViewTabClosingBehaviorButton_Click"
-                                    FontSize="20">TabView Tab Closing behavior Page</Button>
+                            <StackPanel Orientation="Horizontal">
```
Comment (ojhad) [inline dev/TabView/TestUI/TabViewPage.xaml:156]: I think something went wrong with the merge, tabview test page has changes

Comment (RBrid) [reply dev/TabView/TestUI/TabViewPage.xaml:156]: I did it on purpose - it helped getting the gamepad test to pass on my local runs.

**Pull request merged**